projects/packages/search/src/widgets/class-search-widget.php

```diff
@@ -939,7 +947,9 @@ public function render_widget_edit_filter( $filter, $is_template = false ) {
 
 		$args['name_placeholder'] = Helper::generate_widget_filter_name( $args );
 
-		?>
+		// Hide filter author set from Instant Search widget for Classic Search widget
+		if ( $is_instant_search || 'author' !== $args['type'] ) :
```

Comment:
I'm just wondering could we not add the condition here so that it looks a bit cleaner? The switching from Instant Search to Classic Search doesn't work anyway, and which I guess we could address separately 😄

Reply:
If the author filter has been added and then disable Instant Search, the author filter will appear with the wrong type.
<img width="955" alt="截圖 2022-08-16 下午12 45 43" src="https://user-images.githubusercontent.com/6869813/184799794-c5e25520-1041-4a55-b1e3-de9d584ee598.png">